### PR TITLE
Update module to include composer.json and registration.php

### PR DESCRIPTION
**Pull request by dmanners**
* Add in composer.json and registration.php,
* Move files outside of app and update autoloading,

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "davidandvinai/preferencesinfocommand",
```
Comment (Vinai) [inline composer.json]: Should we add the `module-` prefix to the package name, that is `davidandvinai/module-preferencesinfocommand`?
That's the convention the core team uses, I've adopted it for my modules, too. What do you think?

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "description": "Magento 2 bin/magento command to display configured preferences for classes or interfaces",
+    "type": "magento2-module",
+    "license": "BSD-3-Clause",
+    "require": {
+        "php": "~5.6.0|~7.0"
```
Comment (Vinai) [inline composer.json]: Lets drop PHP 5.6 support already. Just because. :)

**Comment by dmanners** [comment]
Not sure if I should add a require on the Magento Framework version. Was cool that it worked on 2.1.4 even though it was written on 0.something.

File: composer.json (hunk 2 above)
Comment (dmanners) [reply composer.json]: sure sure

File: composer.json (hunk 1 above)
Comment (dmanners) [reply composer.json]: I am up for that. what about adding in `-` between words to make it a bit "easier" to read. Not sure myself?

**Pull request merged**